src/manifolds/ShapeSpace.jl

```diff
@@ -0,0 +1,236 @@
+"""
+struct KendallsShapeSpace{n,k} <: AbstractDecoratorManifold{ℝ} end
+
+KendallsShapeSpace(n::Int, k::Int) = KendallsShapeSpace{n,k}()
+
+function active_traits(f, ::KendallsShapeSpace, args...)
+    return merge_traits(IsIsometricEmbeddedManifold(), IsQuotientManifold())
+end
+
+function get_orbit_action(M::KendallsShapeSpace{n,k}) where {n,k}
+    return ColumnwiseMultiplicationAction(M, SpecialOrthogonal(n))
+end
+
+@doc raw"""
+    get_total_space(::Grassmann{n,k})
+
+Return the total space of the [`KendallsShapeSpace`](@ref) manifold, which is the
+[`KendallsPreShapeSpace`](@ref) manifold.
+"""
+get_total_space(::KendallsShapeSpace{n,k}) where {n,k} = KendallsPreShapeSpace(n, k)
+
+function distance(M::KendallsShapeSpace, p, q)
+    A = get_orbit_action(M)
+    a = optimal_alignment(A, p, q)
+    rot_q = apply(A, a, q)
+    return distance(get_embedding(M), p, rot_q)
+end
+
+"""
+    exp(M::KendallsShapeSpace, p, X)
+
+Compute the exponential map on [`KendallsShapeSpace`](@ref) `M`. See [^Guigui2021] for discussion
+about its computation.
+
+[^Guigui2021]:
+    > N. Guigui, E. Maignant, A. Trouvé, and X. Pennec, “Parallel Transport on Kendall Shape
+    > Spaces,” in Geometric Science of Information, Cham, 2021, pp. 103–110.
+    > doi: 10.1007/978-3-030-80209-7_12.
+"""
+exp(M::KendallsShapeSpace, p, X)
+
+function exp!(M::KendallsShapeSpace, q, p, X)
+    return exp!(get_embedding(M), q, p, X)
+end
+
+embed(::KendallsShapeSpace, p) = p
+embed(::KendallsShapeSpace, p, X) = X
+
+function get_embedding(::KendallsShapeSpace{N,K}) where {N,K}
+    return KendallsPreShapeSpace(N, K)
+end
+
+function Base.isapprox(M::KendallsShapeSpace, p, X, Y; atol=sqrt(max_eps(X, Y)), kwargs...)
+    return isapprox(norm(M, p, X - Y), 0; atol=atol, kwargs...)
+end
+function Base.isapprox(M::KendallsShapeSpace, p, q; atol=sqrt(max_eps(p, q)), kwargs...)
+    return isapprox(distance(M, p, q), 0; atol=atol, kwargs...)
+end
+
+"""
+    log(M::KendallsShapeSpace, p, q)
+
+Compute the logarithmic map on [`KendallsShapeSpace`](@ref) `M`. See [^Guigui2021] for discussion
+about its computation.
+
+[^Guigui2021]:
+    > N. Guigui, E. Maignant, A. Trouvé, and X. Pennec, “Parallel Transport on Kendall Shape
+    > Spaces,” in Geometric Science of Information, Cham, 2021, pp. 103–110.
+    > doi: 10.1007/978-3-030-80209-7_12.
+"""
+log(M::KendallsShapeSpace, p, q)
+
+function log!(M::KendallsShapeSpace, X, p, q)
+    A = get_orbit_action(M)
+    a = optimal_alignment(A, p, q)
+    rot_q = apply(A, a, q)
+    return log!(get_embedding(M), X, p, rot_q)
+end
+
+@doc raw"""
+    manifold_dimension(M::KendallsShapeSpace)
+
+Return the dimension of the [`KendallsShapeSpace`](@ref) manifold `M`. The dimension is given by
+``nk - 1 - n(n - 1)/2``
+
+"""
+function manifold_dimension(::KendallsShapeSpace{n,k}) where {n,k}
+    return n * (k - 1) - 1 - div(n * (n - 1), 2)
+end
+
+function project!(::KendallsShapeSpace{n,k}, q, p) where {n,k}
```

Comment:
This function I do not understand isn't it basically calling itself again?

Reply:
No, it's calling projection in the embedding, I probably should've written `get_embedding(M)` instead of `KendallsPreShapeSpace(n, k)` (notice that there is `Pre` in the name).

Reply:
Oh, I missed the `Pre` indeed. Maybe using `get_embedding` is more clear here, yes.